readthedocs/api/v3/serializers.py

```diff
@@ -475,6 +475,10 @@ class Meta:
 
     def validate_name(self, value):
         potential_slug = slugify(value)
+        if not potential_slug:
+            raise serializers.ValidationError(
+                _('Invalid project name "{0}": no slug generated.').format(value),
+            )
```

Comment:
I really wish this used the same logic as our models/form/views that already have this built in. That ship has sailed, but always makes me sad that we have to duplicate this logic.

Reply:
I remember that I worked on this trying to use the same logic everywhere and I think there was a helper function.

... but it seems I'm wrong because I didn't find it with a quick search :/